src/jit/gentree.cpp

```diff
@@ -1442,14 +1442,55 @@ bool GenTree::Compare(GenTree* op1, GenTree* op2, bool swapOK)
                 }
             }
 
-            if (Compare(op1->gtCall.gtCallLateArgs, op2->gtCall.gtCallLateArgs) &&
-                Compare(op1->gtCall.gtCallArgs, op2->gtCall.gtCallArgs) &&
-                Compare(op1->gtCall.gtControlExpr, op2->gtCall.gtControlExpr) &&
-                Compare(op1->gtCall.gtCallObjp, op2->gtCall.gtCallObjp))
+            if (!Compare(op1->AsCall()->gtCallObjp, op2->AsCall()->gtCallObjp))
             {
-                return true;
+                return false;
             }
-            break;
+
+            {
```

Comment:
```
}
else 
{
```
?

Reply:
Hmm, don't know. I usually don't like to add `else` when the `if` branch returns. Besides, the subsequent scoping is only required because we're in a switch. Perhaps I should just pulled out all the comapre code in a `GenTreeCall::Equals` function like I did for PHIs.